BackofficeBundle/Tests/Security/Authorization/Voter/NodeGroupRoleVoterTest.php

```diff
@@ -130,19 +134,93 @@ public function testVote($expectedVoterResponse, $nodeId, $ngrNodeId, $ngrRole,
+            array(VoterInterface::ACCESS_DENIED, 'nodeId', 'nodeId', TreeNodesPanelStrategy::ROLE_ACCESS_CREATE_NODE, "1"),
+            array(VoterInterface::ACCESS_DENIED, 'nodeId', 'otherNodeId', TreeNodesPanelStrategy::ROLE_ACCESS_CREATE_NODE, "1"),
+            array(VoterInterface::ACCESS_DENIED, 'nodeId', 'nodeId', TreeNodesPanelStrategy::ROLE_ACCESS_CREATE_NODE, "0"),
+            array(VoterInterface::ACCESS_DENIED, 'nodeId', 'otherNodeId', TreeNodesPanelStrategy::ROLE_ACCESS_CREATE_NODE, "0"),
+            array(VoterInterface::ACCESS_ABSTAIN, 'nodeId', 'nodeId', TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE, "1", 'fakeSiteId'),
+            array(VoterInterface::ACCESS_ABSTAIN, 'nodeId', 'otherNodeId', TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE, "1", 'fakeSiteId'),
+            array(VoterInterface::ACCESS_ABSTAIN, 'nodeId', 'nodeId', TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE, "0", 'fakeSiteId'),
+            array(VoterInterface::ACCESS_ABSTAIN, 'nodeId', 'otherNodeId', TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE, "0", 'fakeSiteId'),
+            array(VoterInterface::ACCESS_ABSTAIN, 'nodeId', 'nodeId', '5640af7a02b0cf39178b4598', "0", 'fakeSiteId'),
+        );
+    }
+
+    /**
+     * Test Vote with inherit access
+     *
+     * @dataProvider provideResponseAndNodeDataInherit
+     */
+    /**
+     * @param int    $expectedVoterResponse
+     * @param string $nodeId
+     * @param string $ngrNodeId
+     * @param string $ngrRole
+     * @param string $ngrParentGranted
+     * @param string $groupSiteId
+     *
+     * @dataProvider provideResponseAndNodeDataInherit
+     */
+    public function testVoteInherit($expectedVoterResponse, $nodeId,  $ngrNodeId, $ngrRole, $ngrParentGranted, $groupSiteId = 'siteId')
+    {
+        $siteId = 'siteId';
+        $parentId = 'parentId';
+        $role = TreeNodesPanelStrategy::ROLE_ACCESS_UPDATE_NODE;
+
+        $parentNode = Phake::mock('OpenOrchestra\ModelBundle\Document\Node');
+
+        $node = Phake::mock('OpenOrchestra\ModelBundle\Document\Node');
+        Phake::when($node)->getNodeId()->thenReturn($nodeId);
+        Phake::when($node)->getSiteId()->thenReturn($siteId);
+        Phake::when($node)->getParentId()->thenReturn($parentId);
+
+        Phake::when($this->nodeRepository)->findInLastVersion(Phake::anyParameters())->thenReturn($parentNode);
+
+        $nodeGroupRole = Phake::mock('OpenOrchestra\BackofficeBundle\Model\NodeGroupRoleInterface');
+        Phake::when($nodeGroupRole)->getNodeId()->thenReturn($ngrNodeId);
+        Phake::when($nodeGroupRole)->getRole()->thenReturn($ngrRole);
+        Phake::when($nodeGroupRole)->isGranted()->thenReturn("inherit");
+
+        $parentNodeGroupRole = Phake::mock('OpenOrchestra\BackofficeBundle\Model\NodeGroupRoleInterface');
+        Phake::when($nodeGroupRole)->getNodeId()->thenReturn($parentId);
```

Comment:
This line and above should test the properties of `$parentNodeGroupRole` instead of `$nodeGroupRole`.